aiohttp_client_cache/backends/dynamodb.py

```diff
@@ -1,148 +1,179 @@
+import logging
 import pickle
-from typing import Dict, Iterable
+from typing import Dict, Iterable, Optional
 
-import boto3
-from boto3.resources.base import ServiceResource
+import aioboto3
+from aioboto3.session import ResourceCreatorContext
 from botocore.exceptions import ClientError
 
 from aiohttp_client_cache.backends import BaseCache, CacheBackend, ResponseOrKey
 from aiohttp_client_cache.forge_utils import extend_signature
+from aiohttp_client_cache.response import CachedResponse
+
+logger = logging.getLogger(__name__)
+
+# DynamoDB has a 400 KB max item size
+# https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/Limits.html
+MAX_DOC_SIZE = 400 * 1024
 
 
 class DynamoDBBackend(CacheBackend):
     """DynamoDB cache backend.
     See :py:class:`.DynamoDbCache` for backend-specific options
-    See `DynamoDB Service Resource
-    <https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/dynamodb.html#service-resource>`_
-    for more usage details.
 
     See :py:class:`.CacheBackend` for args.
     """
 
     @extend_signature(CacheBackend.__init__)
-    def __init__(self, cache_name: str = 'aiohttp-cache', **kwargs):
+    def __init__(self,
+                 cache_name: str = 'aiohttp-cache',
+                 key_attr_name: str = 'k',
+                 val_attr_name: str = 'v',
+                 create_if_not_exists: bool = False,
+                 context: ResourceCreatorContext = None,
+                 **kwargs):
         super().__init__(cache_name=cache_name, **kwargs)
-        self.responses = DynamoDbCache(cache_name, 'responses', **kwargs)
-        self.redirects = DynamoDbCache(
-            cache_name, 'redirects', connection=self.responses.connection
-        )
+        if not context:
+            context = aioboto3.resource("dynamodb")
+        self.responses = DynamoDbCache(cache_name, 'resp', key_attr_name, val_attr_name, create_if_not_exists, context)
+        self.redirects = DynamoDbCache(cache_name, 'redir', key_attr_name, val_attr_name, create_if_not_exists, context)
+
+    async def get_response(self, key: str) -> Optional[CachedResponse]:
+        logger.debug(f'Attempting to get cached response for key: {key}')
+
+        # Avoiding calling contains here
+        response = await self.responses.read(key)
+        if not response:
+            redirect_key = await self.redirects.read(key)
+            if redirect_key:
+                response = await self.responses.read(redirect_key)
+
+        if not response:
+            logger.debug('No cached response found')
+            return None
+
+        # TODO Ask when might we expect this?
```

Comment:
I wasn't sure why this block was here in the super impl, other than to maybe catch some early problems?

Reply:
Pretty much. After adding the `CachedResponse` wrapper class, previously cached responses would raise an error.

I'd say go ahead and remove that.